### PR TITLE
feat: add Dynatrace datasource

### PR DESCRIPTION
**Pull request by agardnerIT**
This PR adds Dynatrace as a DataStore to Tracetest.

## Changes

- Adds Dynatrace as a datasource

## Checklist

- [] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
+# Running Tracetest With Datadog
```
Comment (kdhamric) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md:1]: I believe the references to Datadog need to be changed to Dynatrace

**Review by kdhamric — COMMENTED** [review]
Marked a couple spots that may need correction

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+
+:::note
+[Check out the source code on GitHub here.](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-datadog)
```
Comment (danielbdias) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md:4]:
```suggestion
[Check out the source code on GitHub here.](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-dynatrace)
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+:::
+
+[Tracetest](https://tracetest.io/) is a testing tool based on [OpenTelemetry](https://opentelemetry.io/) that allows you to test your distributed application. It allows you to use data from distributed traces generated by OpenTelemetry to validate and assert if your application has the desired behavior defined by your test definitions.
+
+[Datadog](https://www.datadoghq.com/) is an observability solution for cloud-scale applications, providing solutions to monitor databases, servers, tools and services. It provides integrated distributed tracing, logs and metrics solutions and supports OpenTelemetry standards.
```
Comment (danielbdias) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md:9]: Here we can change to a specific description for Dynatrace, maybe something like this? (partially inspired by this [Wikipedia description](https://en.wikipedia.org/wiki/Dynatrace))

```suggestion
[Dynatrace](https://www.dynatrace.com/) is an observability solution used to monitor and optimize application performance, software development and security practices, IT infrastructure, and user experience. It provides integrated distributed tracing, logs, and metrics solutions and supports OpenTelemetry standards.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+## OpenTelemetry Demo `v1.3.0` with Datadog, OpenTelemetry and Tracetest
+
+This is a simple sample app on how to configure the [OpenTelemetry Demo `v1.3.0`](https://github.com/open-telemetry/opentelemetry-demo) to use [Tracetest](https://tracetest.io/) for enhancing your E2E and integration tests with trace-based testing and [Datadog](https://www.datadoghq.com/) as a trace data store.
```
Comment (danielbdias) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md:13]:
```suggestion
This is a simple sample app on how to configure the [OpenTelemetry Demo `v1.3.0`](https://github.com/open-telemetry/opentelemetry-demo) to use [Tracetest](https://tracetest.io/) for enhancing your E2E and integration tests with trace-based testing and [Dynatrace](https://www.dynatrace.com/) as a trace data store.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+
+## Prerequisites
+
+You will need [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/) installed on your machine to run this sample app! Additionally, you will need a Datadog account and an API. Sign up to Datadog on their [homepage](https://www.datadoghq.com/) by clicking on `Get Started Free`.
```
Comment (danielbdias) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md:17]:
```suggestion
You will need [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/) installed on your machine to run this sample app! Additionally, you will need a Datadog account and an API Key. Sign up for Dynatrace on their [homepage](https://www.dynatrace.com/) by clicking on `Free trial`.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+
+---
+type: DataStore
+spec:
+  name: datadog
+  type: datadog
+
+```
+
+**Sending Traces to Tracetest and Datadog**
```
Comment (danielbdias) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md:198]:
```suggestion
**Sending Traces to Tracetest and Dynatrace**
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+    # Read more in docs here: https://docs.tracetest.io/configuration/connecting-to-data-stores/opentelemetry-collector
+    tls:
+      insecure: true
+  # Datadog exporter
+  datadog:
+    api:
+      site: datadoghq.com
+      key: ${DATADOG_API_KEY} # Add here you API key for Datadog
+    # One example on how to set up a collector configuration for Datadog can be seen here:
+    # https://docs.datadoghq.com/opentelemetry/otel_collector_datadog_exporter/?tab=onahost
```
Comment (danielbdias) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md:231]: Here we need to change to Dynatrace-specific configurations, probably similar to the setup here: `examples/tracetest-dynatrace/tracetest/collector.config.yaml`

**Review by danielbdias — COMMENTED** [review]
hi @agardnerIT ! thanks for your contribution! 😄 
I've added some comments that we can improve on this PR. Does them make sense?

**Comment by agardnerIT** [comment]
Really not sure how I missed all of that! Hopefully those new commits resolve all the issues.

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md (hunk 5 above)
Comment (kdhamric) [reply docs/docs/examples-tutorials/recipes/running-tracetest-with-dynatrace.md]: @danielbdias is going to review this today. If we can get it knocked out, we may be able to slide it into the RC on Monday and then release on Tuesday!

**Review by danielbdias — COMMENTED** [review]
Added some comments that can help to fix the CI for the server and frontend. By making these changes, we are ready to go with this PR.

**Comment by agardnerIT** [comment]
I've fixed the SVG and code comments. Having some trouble running `make generate-web`.

1. No windows support
2. I run in a codespace and download goreleaser pro 1.18.2 then run `make generate-web` from root and get this:

```
@agardnerIT ➜ /workspaces/tracetest (add-dt) $ make generate-web
make: Nothing to be done for 'generate-web'.
```

3. `make goreleaser-version`:

```
make: Nothing to be done for 'goreleaser-version'.
```

5. `$ make build-web
cd web; npm run build

> web@0.1.0 build
> npm run less && craco build

> web@0.1.0 less
> lessc --js src/antd-theme/antd-customized.less src/antd-theme/antd-customized.css

Creating an optimized production build...
TypeError: Cannot set property mark of #<Object> which has only a getter
TypeError: Cannot set property mark of #<Object> which has only a getter
    at Object.connectTypeScriptPerformance (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/profile/TypeScriptPerformance.js:12:36)
    at createTypeScriptReporter (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/reporter/TypeScriptReporter.js:40:49)
    at Object.<anonymous> (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/reporter/reporter-rpc/ReporterRpcService.js:21:30)
    at Generator.next (<anonymous>)
    at /workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/reporter/reporter-rpc/ReporterRpcService.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/reporter/reporter-rpc/ReporterRpcService.js:4:12)
    at /workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/reporter/reporter-rpc/ReporterRpcService.js:19:88
    at Object.<anonymous> (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/RpcService.js:23:38)
    at Generator.next (<anonymous>)
TypeError: Cannot set property mark of #<Object> which has only a getter
TypeError: Cannot set property mark of #<Object> which has only a getter
    at Object.connectTypeScriptPerformance (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/profile/TypeScriptPerformance.js:12:36)
    at createTypeScriptReporter (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/typescript-reporter/reporter/TypeScriptReporter.js:40:49)
    at Object.<anonymous> (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/reporter/reporter-rpc/ReporterRpcService.js:21:30)
    at Generator.next (<anonymous>)
    at /workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/reporter/reporter-rpc/ReporterRpcService.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/reporter/reporter-rpc/ReporterRpcService.js:4:12)
    at /workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/reporter/reporter-rpc/ReporterRpcService.js:19:88
    at Object.<anonymous> (/workspaces/tracetest/web/node_modules/fork-ts-checker-webpack-plugin/lib/rpc/RpcService.js:23:38)
    at Generator.next (<anonymous>)
```

Side note: It would be great to have a preconfigured codespace w/ batteries included to help dev.

**Comment by danielbdias** [comment]
> Side note: It would be great to have a preconfigured codespace w/ batteries included to help dev.

Thanks for this feedback @agardnerIT ! I will double-check with the team about this.

I will take a look on `generate-web` and fix that for this PR.

**Comment by kdhamric** [comment]
Very excited to have this in and released!!!

**Pull request merged**